src/phy.c

```diff
@@ -338,3 +338,179 @@ static int handle_ed_scan(struct nl802154_state *state,
+		}
+		if (tb_pan_desc[NL802154_ATTR_PAN_DESC_SEC_STATUS]) {
+		    printf("\tSec status   : %x\n", nla_get_u8(tb_pan_desc[NL802154_ATTR_PAN_DESC_SEC_STATUS]));
+		}
+		if (tb_pan_desc[NL802154_ATTR_PAN_DESC_SEC_LEVEL]) {
+		    printf("\tSec level    : %x\n", nla_get_u8(tb_pan_desc[NL802154_ATTR_PAN_DESC_SEC_LEVEL]));
+		}
+		if (tb_pan_desc[NL802154_ATTR_PAN_DESC_KEY_ID_MODE]) {
+		    printf("\tKey Id Mode  : %x\n", nla_get_u8(tb_pan_desc[NL802154_ATTR_PAN_DESC_KEY_ID_MODE]));
+		}
+		if (tb_pan_desc[NL802154_ATTR_PAN_DESC_KEY_SRC]) {
+		    printf("\tKey Src      : %x\n", nla_get_u8(tb_pan_desc[NL802154_ATTR_PAN_DESC_KEY_SRC]));
+		}
+		if (tb_pan_desc[NL802154_ATTR_PAN_DESC_KEY_INDEX]) {
+		    printf("\tKey Index    : %d\n", nla_get_u8(tb_pan_desc[NL802154_ATTR_PAN_DESC_KEY_INDEX]));
+		}
+		goto out;
+
+	} else if( tb_msg[NL802154_ATTR_SCAN_STATUS]){
+		printf( "Active Scan Confirm \n");
+		printf( "\tScan Status: %d\n",nla_get_u8( tb_msg[NL802154_ATTR_SCAN_STATUS] ) );
+		printf( "\tScan Type: %d\n",nla_get_u8( tb_msg[NL802154_ATTR_SCAN_TYPE] ) );
+		printf( "\tAttribute Page: %d\n",nla_get_u8( tb_msg[NL802154_ATTR_PAGE] ) );
+		printf( "\tScan Detect Category: %d\n",nla_get_u8( tb_msg[NL802154_ATTR_SCAN_DETECTED_CATEGORY] ) );
+		printf( "\tScan Result List Size: %d\n",nla_get_u8( tb_msg[NL802154_ATTR_SCAN_RESULT_LIST_SIZE] ) );
+
+	} else {
+		goto protocol_error;
+	}
+
+protocol_error:
+	fprintf( stderr, "protocol error\n" );
+	r = -EINVAL;
+out:
+	return NL_SKIP;
+}
+
+static int handle_active_scan(struct nl802154_state *state,
+               struct nl_cb *cb,
+               struct nl_msg *msg,
+               int argc, char **argv,
+               enum id_input id)
+{
+	int r;
+	int i;
+
+	const uint8_t scan_type = IEEE802154_MAC_SCAN_ACTIVE;
+	uint32_t channel_page = 0;
+	static uint32_t scan_channels;
+	uint32_t scan_duration;
+
+	if ( argc >= 1 ) {
+		if ( 1 != sscanf( argv[ 0 ], "%u", &channel_page ) ) {
+			goto invalid_arg;
+		}
+	}
+	if ( argc >= 2 ) {
+		if ( ! (
+				( 0 == strncmp( "0x", argv[ 1], 2 ) && 1 == sscanf( argv[ 1 ] + 2, "%x", &scan_channels ) ) ||
+				1 == sscanf( argv[ 1 ], "%u", &scan_channels )
+		) ) {
+			goto invalid_arg;
+		}
+	}
+	if ( argc == 3 ) {
+		if ( 1 != sscanf( argv[ 2 ], "%u", &scan_duration ) ) {
+			goto invalid_arg;
+		}
+	}
+	if ( argc > 3 ) {
+		goto invalid_arg;
+	}
+
+	NLA_PUT_U8(msg, NL802154_ATTR_SCAN_TYPE, scan_type);
+	NLA_PUT_U32(msg, NL802154_ATTR_SUPPORTED_CHANNEL, scan_channels );
+	NLA_PUT_U8(msg, NL802154_ATTR_SCAN_DURATION, scan_duration);
+	NLA_PUT_U8(msg, NL802154_ATTR_PAGE, channel_page);
+
+	r = nl_cb_set(cb, NL_CB_VALID, NL_CB_CUSTOM, print_active_scan_results, &scan_channels );
+	if ( 0 != r ) {
+		goto out;
+	}
+
+	r = 0;
+	goto out;
+
+	nla_put_failure:
+	r = -ENOBUFS;
+	out:
+	return r;
+	invalid_arg:
+	r = 1;
+	goto out;
+}
+COMMAND(set, active_scan, "<channel_page> <scan channel bitmask> <scan duration>",
```

Comment:
can you change this to a 'get' command. Not a huge deal, but just a bit more intuitive